tests/keras_tests/infrastructure_tests/base_keras_infrastructure_test.py

```diff
@@ -0,0 +1,113 @@
+# limitations under the License.
+# ==============================================================================
+from typing import Dict
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import TensorShape
+from tensorflow_model_optimization.python.core.quantization.keras.quantize_wrapper import QuantizeWrapper
+
+from model_compression_toolkit import quantizers_infrastructure as qi, QuantizationConfig
+from model_compression_toolkit.core.common.quantization.node_quantization_config import NodeWeightsQuantizationConfig, \
+    NodeActivationQuantizationConfig
+from model_compression_toolkit.core.common.target_platform import OpQuantizationConfig, QuantizationMethod
+
+keras = tf.keras
+layers = keras.layers
+
+DISPATCHER = "dispatcher"
+CLASS_NAME = "class_name"
+
+
+class IdentityQuantizer(qi.BaseKerasTrainableQuantizer):
+    def __init__(self, quantization_config: NodeWeightsQuantizationConfig):
+        super().__init__(quantization_config,
+                         qi.QuantizationTarget.Weights,
+                         [qi.QuantizationMethod.POWER_OF_TWO, qi.QuantizationMethod.SYMMETRIC])
+
+    def initialize_quantization(self,
+                                tensor_shape: TensorShape,
+                                name: str,
+                                layer: QuantizeWrapper) -> Dict[str, tf.Variable]:
+        return
+
+    def __call__(self,
+                 inputs: tf.Tensor,
+                 training: bool):
+        return inputs
+
+
+class ZeroActivationsQuantizer(qi.BaseKerasTrainableQuantizer):
+    def __init__(self, quantization_config: NodeActivationQuantizationConfig):
+        super().__init__(quantization_config,
+                         qi.QuantizationTarget.Activation,
+                         [qi.QuantizationMethod.POWER_OF_TWO, qi.QuantizationMethod.SYMMETRIC])
+
+    def initialize_quantization(self,
+                                tensor_shape: TensorShape,
+                                name: str,
+                                layer: QuantizeWrapper) -> Dict[str, tf.Variable]:
+        return
+
+    def __call__(self,
+                 inputs: tf.Tensor,
+                 training: bool = True) -> tf.Tensor:
+        return inputs * 0
+
+
+def dummy_fn():
+    return
+
+
+op_cfg = OpQuantizationConfig(QuantizationMethod.POWER_OF_TWO,
+                              QuantizationMethod.POWER_OF_TWO,
+                              8,
+                              8,
+                              True,
+                              True,
+                              True,
+                              True,
+                              1,
+                              0,
+                              32)
+qc = QuantizationConfig()
+weight_quantization_config = NodeWeightsQuantizationConfig(qc, op_cfg, dummy_fn, dummy_fn, -1)
+activation_quantization_config = NodeActivationQuantizationConfig(qc, op_cfg, dummy_fn, dummy_fn)
+
+
+class BaseKerasInfrastructureTest:
```

Comment:
I think the new class should be based on  unittest.TestCase - 
class BaseKerasInfrastructureTest (unittest.TestCase):